options/options.go

```diff
@@ -107,6 +108,15 @@ func New() (*Options, error) {
 	}
 
 	opts.SetPublishResults()
+
+	opts.ScorecardOpts.ResultsFile = os.Getenv(EnvInputResultsFile)
+
+	// Set PAT if not provided (locally).
+	_, repoTokenExists := os.LookupEnv(EnvGithubAuthToken)
```

Comment:
in which scenario is the EnvGithubAuthToken set and not set? It's set by GitHub, so we don't control it, right?

Note that we could set the default PAT to be the GH token in this file https://github.com/ossf/scorecard-action/blob/main/action.yaml#L30

Reply:
Looks like there was a redundant env var (GITHUB_AUTH_TOKEN) being set. fixed this so it directly uses the input.